rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/grants/code/JPACodeDataProvider.java

```diff
@@ -143,13 +143,13 @@ public void setCodeLifetime(long codeLifetime) {
         if (c == null && resourceOwnerSubject == null) {
             return em.createQuery("SELECT c FROM ServerAuthorizationCodeGrant c",
                     ServerAuthorizationCodeGrant.class);
-        } else if (c == null) {
+        } else if (c == null && resourceOwnerSubject != null) {
```

Comment:
I'm not sure if this change is needed?

Reply:
there are 4 null/non-null combinations possible; added explicit check for all cases